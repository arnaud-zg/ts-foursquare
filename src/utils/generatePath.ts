import pathToRegexp from 'path-to-regexp'

const compilePath = (path: string) => pathToRegexp.compile(path)

/**
 * Public API for generating a URL pathname from a path and parameters.
 * https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/generatePath.js#L20
 */
export const generatePath = (path = '/', params = {}) =>
  path === '/' ? path : compilePath(path)(params)
