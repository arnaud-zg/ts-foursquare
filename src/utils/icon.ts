import { NIcon } from '../../types/icon.d'
import { EIconSize } from '../constants/icon'

interface IIconUrls {
  [iconSize: string]: string
}

export const getIconUrlList = ({ prefix, suffix }: NIcon.IIcon): IIconUrls =>
  Object.keys(EIconSize).reduce(
    (acc, iconKey) => ({
      ...acc,
      [iconKey]: `${prefix}${
        // Ugly typescript hack
        (EIconSize as { [key: string]: any })[iconKey]
      }${suffix}`,
    }),
    {}
  )
