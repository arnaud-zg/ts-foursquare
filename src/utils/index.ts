export * from './icon'
export * from './url'
