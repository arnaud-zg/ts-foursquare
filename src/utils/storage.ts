import store from 'store'
import { PROJECT_NAME } from '../constants'
import { CLIENT_ID_KEY, CLIENT_SECRET_KEY } from '../constants/api'

export const getCredentials = () => {
  const data = store.get(PROJECT_NAME) || {}

  const { [CLIENT_ID_KEY]: clientId, [CLIENT_SECRET_KEY]: clientSecret } = data

  return {
    clientId,
    clientSecret,
  }
}

export const setCredentials = ({
  clientId,
  clientSecret,
}: {
  clientId: string
  clientSecret: string
}) => {
  if (clientId && clientSecret) {
    store.set(PROJECT_NAME, {
      [CLIENT_ID_KEY]: clientId,
      [CLIENT_SECRET_KEY]: clientSecret,
    })
  }
}

export const removeCredentials = () => {
  store.remove(PROJECT_NAME)
}
