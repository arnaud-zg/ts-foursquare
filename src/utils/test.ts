import { NHttpStatuses } from 'http-response-status'

export const mockingFetch = ({ response = {} } = {}) => {
  const mockJsonPromise = Promise.resolve(response)
  const mockFetchPromise = Promise.resolve({
    json: () => mockJsonPromise,
    ok: true,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotValidFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: true,
    json: null,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotFoundFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: false,
    status: NHttpStatuses.EClientError.NOT_FOUND,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}
