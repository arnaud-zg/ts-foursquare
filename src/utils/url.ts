interface ILocationSearchParam {
  [key: string]: string | number | boolean | undefined
}

export const getLocationSearch = (
  { param = {} }: { param: ILocationSearchParam } = { param: {} }
) => {
  const locationSearch = Object.keys(param)
    .filter(key => param[key] !== undefined && param[key] !== '')
    .map(
      key => `${encodeURIComponent(key)}=${encodeURIComponent(`${param[key]}`)}`
    )
    .join('&')

  return (!!locationSearch && `?${locationSearch}`) || ''
}

export const getLocationHref = ({
  origin,
  pathname,
  param = {},
}: {
  origin: string
  pathname: string
  param?: ILocationSearchParam
}) => `${origin}${pathname}${getLocationSearch({ param })}`
