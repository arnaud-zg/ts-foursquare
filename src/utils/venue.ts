import { NVenueListed } from '../../types'

export const getVenuesListedGroupKey = (group: NVenueListed.IGroup) =>
  `${group.type}-${group.name}`
