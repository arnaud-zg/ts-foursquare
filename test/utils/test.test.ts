import { mockingFetch } from '../../src/utils/test'

const data = {
  text: 'Hello World',
}

describe('mockingFetch()', () => {
  mockingFetch({
    response: {
      ...data,
    },
  })

  const fetchResult = fetch('https://api.tld.com/v2/randomEndpoint')

  it('should get a promise', () => {
    expect(fetchResult).toMatchSnapshot()
  })

  it('should get a success Response object', async () => {
    const response = await fetchResult
    expect(response).toMatchSnapshot()
  })

  it('should get response', async () => {
    const response = await fetchResult
    const data = await response.json()
    expect(data).toEqual({
      ...data,
    })
  })
})
