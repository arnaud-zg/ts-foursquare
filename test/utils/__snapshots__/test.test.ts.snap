// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mockingFetch() should get a promise 1`] = `Promise {}`;

exports[`mockingFetch() should get a success Response object 1`] = `
Object {
  "json": [Function],
  "ok": true,
}
`;
